utils/android-push.sh

```diff
@@ -11,15 +11,32 @@ fi
 
 # Push to Android Firefox if device is connected
 # XXX on some systems, adb may require sudo...
-if type adb > /dev/null 2>/dev/null && adb devices >/dev/null 2>/dev/null ; then
-  ADB_FOUND=`adb devices | grep -v 'offline$' | tail -2 | head -1 | cut -f 1 | sed 's/ *$//g'`
-  if [ "$ADB_FOUND" != "List of devices attached" ]; then
-    echo Pushing "$XPI_NAME" to /sdcard/"$XPI_NAME"
-    adb push "../$XPI_NAME" /sdcard/"$XPI_NAME"
-    adb shell am start -a android.intent.action.VIEW \
-                       -c android.intent.category.DEFAULT \
-                       -d file:///mnt/sdcard/pkg/ \
-                       -n $ANDROID_APP_ID/.App
+if type adb > /dev/null 2>/dev/null ; then
+  #running `adb devices` below will start adb server/daemon if it wasn't running already
+  #we start it and make note if it's us that started it or not, so we can stop it afterwards
+  if adb start-server 2>&1 |grep -qF 'daemon not running' ; then
+    we_started_adb_daemon=1
+  else
+    we_started_adb_daemon=0
+  fi
+  on_exit() {
+    if test "0$we_started_adb_daemon" == "01"; then
```

Comment:
Why the `0` prefix here?

Reply:
This code didn't make any sense to me either.

Reply:
> Why the `0` prefix here?

It is clearly not necessary in this particular case, however I was merely trying to future proof it just in case in the future `==` would become `-eq`(or any other arithmetic operator) **and** the variable (`$we_started_adb_daemon`) would be empty/unset. Unlikely, but still, that was the idea.

Reply:
Thanks.  Let's leave it out, it'll be the responsibility of whoever opens that PR to add it back.  Ready to merge otherwise.

Reply:
> Thanks. Let's leave it out, it'll be the responsibility of whoever opens that PR to add it back. Ready to merge otherwise.

All done.